src/common/internet.rs

```diff
@@ -0,0 +1,364 @@
+// LNP/BP Rust Library
+// Written in 2020 by
+//     Dr. Maxim Orlovsky <orlovsky@pandoracore.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the MIT License
+// along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+
+///! Universal addresses that support IPv4, IPv6 and Tor
+
+use std::fmt;
+use std::convert::TryFrom;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+#[cfg(feature="use-tor")]
+use torut::onion::{TorPublicKeyV3, OnionAddressV3};
```

Comment:
Why only V3?

Reply:
Because V2 are obsolete already?

Reply:
They are not as good on the privacy level but they are still widely used. I think we should support them if we are trying to do something generic

Reply:
Hm, do they have the same security assumptions?

Basically, by using only V3 in this library we may try to promote usage of V3 and de-usage of V2, if this makes sense...